tests/clvm/test_puzzle_compression.py

```diff
@@ -0,0 +1,105 @@
+from chia.wallet.cc_wallet.cc_utils import CC_MOD, construct_cc_puzzle
+from chia.wallet.puzzles.p2_delegated_puzzle_or_hidden_puzzle import puzzle_for_pk
+
+ZERO_32 = bytes32([0] * 32)
+ONE_32 = bytes32([17] * 32)
+COIN = Coin(ZERO_32, ZERO_32, uint64(0))
+SOLUTION = Program.to([])
+
+
+class TestPuzzleCompression:
+    compression_factors = {}
+
+    def test_standard_puzzle(self):
+        coin_spend = CoinSpend(
+            COIN,
+            puzzle_for_pk(G1Element()),
+            SOLUTION,
+        )
+        compressed = CompressedCoinSpend.compress(coin_spend, LATEST_VERSION)
+        assert len(bytes(coin_spend)) > len(bytes(compressed))
+        assert coin_spend == compressed.decompress()
+        self.compression_factors["standard_puzzle"] = len(bytes(compressed)) / len(bytes(coin_spend))
+
+    def test_cat_puzzle(self):
+        coin_spend = CoinSpend(
+            COIN,
+            construct_cc_puzzle(CC_MOD, Program.to([]).get_tree_hash(), Program.to(1)),
+            SOLUTION,
+        )
+        compressed = CompressedCoinSpend.compress(coin_spend, LATEST_VERSION)
+        assert len(bytes(coin_spend)) > len(bytes(compressed))
+        assert coin_spend == compressed.decompress()
+        self.compression_factors["cat_puzzle"] = len(bytes(compressed)) / len(bytes(coin_spend))
+
+    def test_offer_puzzle(self):
+        coin_spend = CoinSpend(
+            COIN,
+            OFFER_MOD,
+            SOLUTION,
+        )
+        compressed = CompressedCoinSpend.compress(coin_spend, LATEST_VERSION)
+        assert len(bytes(coin_spend)) > len(bytes(compressed))
+        assert coin_spend == compressed.decompress()
+        self.compression_factors["offer_puzzle"] = len(bytes(compressed)) / len(bytes(coin_spend))
+
+    def test_nesting_puzzles(self):
+        coin_spend = CoinSpend(
+            COIN,
+            construct_cc_puzzle(CC_MOD, Program.to([]).get_tree_hash(), puzzle_for_pk(G1Element())),
+            SOLUTION,
+        )
+        compressed = CompressedCoinSpend.compress(coin_spend, LATEST_VERSION)
+        assert len(bytes(coin_spend)) > len(bytes(compressed))
+        assert coin_spend == compressed.decompress()
+        self.compression_factors["cat_w_standard_puzzle"] = len(bytes(compressed)) / len(bytes(coin_spend))
+
+    def test_unknown_wrapper(self):
+        unknown = Program.to([2, 2, []])  # (a 2 ())
+        coin_spend = CoinSpend(
+            COIN,
+            unknown.curry(puzzle_for_pk(G1Element())),
+            SOLUTION,
+        )
+        compressed = CompressedCoinSpend.compress(coin_spend, LATEST_VERSION)
+        assert len(bytes(coin_spend)) > len(bytes(compressed))
+        assert coin_spend == compressed.decompress()
+        self.compression_factors["unknown_and_standard"] = len(bytes(compressed)) / len(bytes(coin_spend))
+
+    def test_lowest_compatible_version(self):
+        assert lowest_compatible_version([bytes(CC_MOD)]) == 1
```

Comment:
This doesn't really make sense; compatibility is not a property of a `Program`. It's a property of the compressed offer blob.